packages/ia-components/sandbox/audio-players/youtube-player.jsx

```diff
@@ -69,15 +77,81 @@ class YoutubePlayer extends Component {
    * Load video of passed id and default resolution
    */
   componentDidUpdate(prevProps) {
+    const { selectedTrack: capturedTrack } = this.state;
     const { id, selectedTrack } = this.props;
-    const trackChanged = id !== prevProps.id;
-
-    if (trackChanged) {
+    const { id: prevId } = prevProps;
+    const trackChanged = id !== prevId;
+    const playerNotStarted = capturedTrack === null;
+    const setID = !trackChanged && playerNotStarted;
+    if (trackChanged || setID) {
       clearTimeout(this.timer);
       this.setState({ id, selectedTrack }, this.playVideo);
     }
   }
 
+  /**
+   * Remove timer & interval
+   */
+  componentWillUnmount() {
+    clearInterval(this.fullAlbumVideoPoller);
+    clearTimeout(this.timer);
+  }
+
+  /**
+   * Gets available YouTube ID
+   * checks between captured ID (state) & incoming ID (props)
+   * when there isn't a captured ID, it means the player has loaded & user has not selected a video
+   */
+  availableVideoId() {
+    const { id: stateID } = this.state;
+    const { id: propsID } = this.props;
+    const availableID = Number.isInteger(stateID) ? stateID : propsID;
+    return availableID;
+  }
+
+  /**
+   * Check player elapsed time & the track it's on
+   *
+   * @return { object } trackValue
+   * trackValue.elapsedTime = number
+   * trackValue.currentTrack = object (return from fullAlbumDetails)
+   */
+  checkTimeAndTrack() {
+    const { fullAlbumDetails, player } = this.state;
+    const elapsedTime = player.getCurrentTime();
+    const currentTrack = chain(fullAlbumDetails)
```

Comment:
**Suggestion**: since `filter` returns a new array and we don't need it intact after this statement, we can just `fullAlbumDetails.filter().pop()` to get the last matched item and drop the lodash dependency.

Reply:
`fullAlbumDetails` is actually a hash with hash values. I didn't want to roll out my own filtering solution, or should we? @shaneriley